shinydashboard/lantern/functions/endpoints.R

```diff
@@ -1,65 +1,112 @@
 # Functions to compute metrics on endpoints
+library(purrr)
 
-get_endpoint_totals <- function(fhir_endpoints_con,fhir_endpoints_info_con) {
-  # Get the table of fhir endpoints. There may be endpoints we have not reached
-  # so get counts of indexed and non-indexed endpoints
-  fhir_endpoints_tbl <- as_tibble(fhir_endpoints_con)
-  fhir_endpoints_info_tbl <- as_tibble(fhir_endpoints_info_con)
+# Will need scalable solution for creating short names from Vendor names for UI
+vendor_short_names <- data.frame(
+  vendor_name = c("Allscripts", "CareEvolution, Inc.", "Cerner Corporation", "Epic Systems Corporation", "Medical Information Technology, Inc. (MEDITECH)", "Unknown"),
+  short_name = c("Allscripts", "CareEvolution", "Cerner", "Epic", "MEDITECH", "Unknown"),
+  stringsAsFactors = FALSE)
+
+# Get Endpoint Totals
+# Return list of counts of:
+# - all registered endpoints
+# - indexed endpoints that have been queried
+# - non-indexed endpoints yet to be queried
+get_endpoint_totals_list <- function(db_tables) {
+  all <- db_tables$fhir_endpoints %>% count() %>% pull(n)
+  indexed <- db_tables$fhir_endpoints_info %>% count() %>% pull(n)
   fhir_endpoint_totals <- list(
-    "all_endpoints"     = nrow(fhir_endpoints_tbl),
-    "indexed_endpoints" = nrow(fhir_endpoints_info_tbl %>% filter(http_response != 0)),
-    "nonindexed_endpoints" = nrow(fhir_endpoints_tbl) - nrow(fhir_endpoints_info_tbl %>% filter(http_response != 0))
+    "all_endpoints"     = all,
+    "indexed_endpoints" = indexed,
+    "nonindexed_endpoints" = max(all - indexed, 0)
```

Comment:
The nonindexed endpoints should probably be calculated differently than just the fhir_endpoints count minus the fhir_endpoints_info count, since the fhir_endpoints table contains a lot of duplicate entries that the fhir_endpoints_info table does not, and so the nonindexed endpoints value is going to be much higher than it actually is. There should only be around 6 non indexed endpoints, but the nonindexed endpoints value when I run it says there are 375, which is due to the duplicate entries in the fhir_endpoints table. If you were able to do a distinct url count that would probably fix the problem

Reply:
Will base endpoint count on distinct(url) from fhir_endpoints table